p2p/libp2p/facade/netMessengersFacade.go

```diff
@@ -0,0 +1,547 @@
+}
+
+// RegisterMessageProcessorForMessenger registers a message process on a topic on the provided messenger
+func (facade *networkMessengersFacade) RegisterMessageProcessorForMessenger(messengerType p2p.NetworkMessengerType, topic string, identifier string, handler p2p.MessageProcessor) error {
+	messenger, found := facade.messengers[messengerType]
+	if !found {
+		log.Warn("calling RegisterMessageProcessor on an unknown messenger, returning error...")
+		return fmt.Errorf("%w of provided type %s", p2p.ErrUnknownMessenger, messengerType)
+	}
+
+	return messenger.RegisterMessageProcessor(topic, identifier, handler)
+}
+
+// UnregisterAllMessageProcessors will unregister all message processors for topics on the main messenger
+func (facade *networkMessengersFacade) UnregisterAllMessageProcessors() error {
+	return facade.UnregisterAllMessageProcessorsForMessenger(p2p.RegularMessenger)
+}
+
+// UnregisterAllMessageProcessorsForMessenger will unregister all message processors for topics on the provided messenger
+func (facade *networkMessengersFacade) UnregisterAllMessageProcessorsForMessenger(messengerType p2p.NetworkMessengerType) error {
+	messenger, found := facade.messengers[messengerType]
+	if !found {
+		log.Warn("calling UnregisterAllMessageProcessors on an unknown messenger, returning error...")
+		return fmt.Errorf("%w of provided type %s", p2p.ErrUnknownMessenger, messengerType)
+	}
+
+	return messenger.UnregisterAllMessageProcessors()
+}
+
+// UnregisterMessageProcessor unregisters a message processes on a topic on the main messenger
+func (facade *networkMessengersFacade) UnregisterMessageProcessor(topic string, identifier string) error {
+	return facade.UnregisterMessageProcessorForMessenger(p2p.RegularMessenger, topic, identifier)
+}
+
+// UnregisterMessageProcessorForMessenger unregisters a message processes on a topic on the provided messenger
+func (facade *networkMessengersFacade) UnregisterMessageProcessorForMessenger(messengerType p2p.NetworkMessengerType, topic string, identifier string) error {
+	messenger, found := facade.messengers[messengerType]
+	if !found {
+		log.Warn("calling UnregisterMessageProcessor on an unknown messenger, returning error...")
+		return fmt.Errorf("%w of provided type %s", p2p.ErrUnknownMessenger, messengerType)
+	}
+
+	return messenger.UnregisterMessageProcessor(topic, identifier)
+}
+
+// Broadcast tries to send a byte buffer onto a topic using the topic name as channel on the main messenger
+func (facade *networkMessengersFacade) Broadcast(topic string, buff []byte) {
+	facade.broadcastOnMessenger(p2p.RegularMessenger, topic, buff)
+}
+
+func (facade *networkMessengersFacade) broadcastOnMessenger(messengerType p2p.NetworkMessengerType, topic string, buff []byte) {
+	messenger, found := facade.messengers[messengerType]
+	if !found {
+		log.Warn("calling Broadcast on an unknown messenger, returning...")
+		return
+	}
```

Comment:
decided to go with the unexported method just in case we'll need to export it later on(see example of CreateTopic and CreateTopicForMessenger). For example, we may need BroadcastOnMessenger for light clients